src/main/java/slimeknights/tconstruct/smeltery/recipe/CastingBasinRecipe.java

```diff
@@ -0,0 +1,26 @@
+package slimeknights.tconstruct.smeltery.recipe;
+
+import net.minecraft.item.ItemStack;
+import net.minecraft.item.crafting.IRecipeSerializer;
+import net.minecraft.item.crafting.Ingredient;
+import net.minecraft.util.ResourceLocation;
+import net.minecraftforge.fluids.FluidStack;
+import slimeknights.tconstruct.smeltery.TinkerSmeltery;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+public class CastingBasinRecipe extends AbstractCastingRecipe {
+
+  public CastingBasinRecipe(ResourceLocation idIn, String groupIn, @Nullable Ingredient ingredient, @Nonnull FluidStack fluidIn, ItemStack result, int coolingTime, boolean consumed, boolean switchSlots) {
+    super(TinkerSmeltery.basinRecipeType, idIn, groupIn, ingredient, fluidIn, result, coolingTime, consumed, switchSlots);
+  }
+
+  public ItemStack getIcon() {
+    return new ItemStack(TinkerSmeltery.castingBasin);
```

Comment:
I feel we should cache this somewhere. Is this our own method, or overridden?

Reply:
From the parent. Honestly, I don't think this is needed, if we use the isDynamic=true
to keep the casting recipes out of the vanilla book I don't think its used.

Reply:
I looked at the vanilla stuff, and say no cache is needed. Even though these are not used, might as well keep them in I guess.